src/ess_rhat.jl

```diff
@@ -566,17 +562,21 @@ function _ess_rhat(
         τ = max(0, 2 * sum_pₜ + max(0, ρ_even) - 1)
 
         # estimate the effective sample size
-        ess[i] = min(ntotal / τ, ess_max)
+        essi[] = min(ntotal / τ, ess_max)
     end
 
-    return (; ess, rhat)
+    if ndims(ess) == 0
+        return (; ess=ess[], rhat=rhat[])
+    else
+        return (; ess, rhat)
+    end
```

Comment:
No this is not orthogonal or breaking. Since we only supported 3D array inputs previously,  `ess` and `rhat` were always vectors, so we never needed this check. But in this PR we also support vector and matrix inputs, representing a single param from 1 or more chains, in which case `ess` and `rhat` are 0-dimensional. And I think it makes more sense to return a scalar in that case than a 0-dimensional array.